addons/sale/models/account_move.py

```diff
@@ -32,12 +32,15 @@ def unlink(self):
             downpayment_lines.unlink()
         return res
 
-    @api.depends('invoice_user_id')
+    @api.depends('invoice_user_id', 'partner_id')
     def _compute_team_id(self):
         for move in self:
-            if not move.invoice_user_id.sale_team_id or not move.is_sale_document(include_receipts=True):
+            if not move.invoice_user_id.sale_team_id or not move.is_sale_document(include_receipts=True) and not move.partner_id.team_id.id:
```

Comment:
Why don't you recompute the sales team if the invoice partner doesn't have one ? :thinking: